fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -252,13 +252,27 @@ def build_upload_command(username, password, source = "/tmp", provider_short_nam
         ("-p #{password.shellescape}" unless use_api_key),
         ("--apiKey #{api_key[:key_id]}" if use_api_key),
         ("--apiIssuer #{api_key[:issuer_id]}" if use_api_key),
+        ("--asc-provider #{provider_short_name}" if !use_api_key && !provider_short_name.to_s.empty?),
```

Comment:
Not that it really matter, but I'm expecting `rubocop` linter to complain and suggest changing `if !` to `unless` here 🤔 
```suggestion
        ("--asc-provider #{provider_short_name}" unless use_api_key || provider_short_name.to_s.empty?),
```

Reply:
rubocop did say nothing about it but i learned unless is better, let me fix it

Reply:
done! https://github.com/fastlane/fastlane/pull/20631/commits/cd2c504c7ba16769aadc2dd9ab465a177c263a9b